Updated for scripts for dGPU/iGPU

hiteshkumar-nv opened the pull request:
1. Added Module.symvers path in Kbuild file.
2. Updated Readme to build and verify the repo.

hiteshkumar-nv committed:
Changes for IGX with dGPU on GA 1.0 release

Need to follow below steps.
- Build /usr/src/nvidia-535.171.04 using sudo make command.
- This will generate Module.symvers
- Provide Module.symvers path in Kbuild file using this patch.

hiteshkumar-nv committed:
Update README.md for IGX GA release 1.0

hiteshkumar-nv committed:
Updated for IGX iGPU GA release 1.0

- Added new script to build driver for IGX and iGPU.
- Update Kbuild to pass include file and Module symbol.

hiteshkumar-nv committed:
Update README.md for iGPU IGX GA 1.0 release.

Added steps to build the code for iGPU IGX GA 1.0 release.

pateldipen1984-nv commented on kernel-module/Kbuild:
```diff
@@ -15,8 +15,13 @@ ifdef NV_BUILD_DGPU
 else
-	KBUILD_EXTRA_SYMBOLS=/usr/src/nvidia/nvidia-oot/Module.symvers
-	ccflags-y += -I/usr/src/nvidia/nvidia-oot/include/
+	ifdef NV_BUILD_IGX
```
why this? Just update the path. And put Note in the readme or here about what possibly end user has to search?

hiteshkumar-nv replied:
Removed this changes.

pateldipen1984-nv commented on kernel-module/build-for-igx-igpu_native.sh:
```diff
@@ -0,0 +1,38 @@
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+export NV_BUILD_IGX=1
```
No need to add this.

hiteshkumar-nv replied:
Remove entire script and update build-for-jetson-drive-igpu-native.sh to search path

hiteshkumar-nv committed:
Updated script for iGPU.

- Remove build-for-igx-igpu_native.sh script.
- Updated build-for-jetson-drive-igpu-native.sh

hiteshkumar-nv committed:
Update README.md

- Updated readme as per scripts.

hiteshkumar-nv committed:
Update README.md

updated application troubleshooting section.

hiteshkumar-nv committed:
Update README.md

remove unused utils.

pateldipen1984-nv commented on kernel-module/Kbuild:
```diff
@@ -13,9 +13,10 @@ ifdef NV_BUILD_DGPU
 	KBUILD_CFLAGS += \
 		-I$(NVIDIA_SRC_DIR) \
 		-DNV_BUILD_DGPU
+	KBUILD_EXTRA_SYMBOLS=/usr/src/nvidia-535.171.04/Module.symvers
```
what if this is different version? can you make generic?

hiteshkumar-nv replied:
This path will change as per release so user need to update it.

pateldipen1984-nv commented on README.md:
```diff
@@ -252,6 +252,22 @@ cd /path/to/this/project/kernel-module/
+sudo apt update
+sudo apt install build-essential bc
+sudo apt install nvidia-cuda-toolkit (Required to build application.)
+cd /usr/src/nvidia-535.171.04 (Plese update this path as per release)
```
how do you get to this path?

hiteshkumar-nv replied:
This path will change as per release so user need to update it.

pateldipen1984-nv commented on kernel-module/build-for-jetson-drive-igpu-native.sh:
```diff
@@ -20,4 +20,26 @@
 
+SRC_DIR="$(find /usr/src/nvidia/nvidia-oot/* -name nv-p2p.h 2>/dev/null|head -1|xargs dirname 2>/dev/null)"
+if [ ! -d "${SRC_DIR}" ]; then
+	SRC_DIR="$(find /usr/src/./linux-headers-tegra-oot* -name nv-p2p.h 2>/dev/null|head -1|xargs dirname 2>/dev/null)"
```
what if you do not find SRC_DIR here as well? there is no check after that.

pateldipen1984-nv replied:
or may be line 29 checks it implicitly I think.

pateldipen1984-nv commented on kernel-module/build-for-jetson-drive-igpu-native.sh:
```diff
@@ -20,4 +20,26 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
 
+SRC_DIR="$(find /usr/src/nvidia/nvidia-oot/* -name nv-p2p.h 2>/dev/null|head -1|xargs dirname 2>/dev/null)"
```
here we want to find the location of the directory that contains these nv-p2p.h and module symbols in doing so there are may be two places i.e. line 23 and line 25. Can we check those places once and for all at the start so you do not have to find it again like in line 34 and 35? If you do that, it will simplify the code.

hiteshkumar-nv replied:
I have optimized the build script for iGPU where we find first correct kernel path and then we are checking for header and module symbols.

hiteshkumar-nv committed:
Optimized build script for iGPU.

pateldipen1984-nv commented on kernel-module/build-for-jetson-drive-igpu-native.sh:
```diff
@@ -20,4 +20,22 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
 
+KERNEL_SRC_DIR="/usr/src/nvidia/nvidia-oot/"
```
This is not kernel src but nvdia-oot modules installation path.

hiteshkumar-nv replied:
Let me update the variable name.

hiteshkumar-nv replied:
Updated variable name.

pateldipen1984-nv commented on kernel-module/build-for-jetson-drive-igpu-native.sh:
```diff
@@ -20,4 +20,22 @@
+	echo "Kernel source not found in $KERNEL_SRC_DIR"
+	KERNEL_SRC_DIR="/usr/src/./linux-headers-tegra-oot*"
+	if [ ! -d ${KERNEL_SRC_DIR} ]; then
+		echo "Kernel source not found in $KERNEL_SRC_DIR"
```
this is wrong message.

hiteshkumar-nv replied:
Updated message as Nvidia OOT  source

pateldipen1984-nv commented on kernel-module/build-for-jetson-drive-igpu-native.sh:
```diff
@@ -20,4 +20,22 @@
+		exit;
+	fi
+fi
+echo "Kernel source ${KERNEL_SRC_DIR}"
```
THis is nvidi oot source so has to be echoed the same.

hiteshkumar-nv replied:
same updated message Nvidia OOT  source

pateldipen1984-nv commented on kernel-module/build-for-jetson-drive-igpu-native.sh:
```diff
@@ -20,4 +20,22 @@
+KERNEL_SRC_DIR="/usr/src/nvidia/nvidia-oot/"
+if [ ! -d ${KERNEL_SRC_DIR} ]; then
+	echo "Kernel source not found in $KERNEL_SRC_DIR"
+	KERNEL_SRC_DIR="/usr/src/./linux-headers-tegra-oot*"
```
I wonder if p2p.h or module symver be ever in this path?

hiteshkumar-nv replied:
Yes both are at same directory for Nvidia OOT.
/usr/src/nvidia/nvidia-oot/include/linux/nv-p2p.h
/usr/src/nvidia/nvidia-oot/Module.symvers

pateldipen1984-nv replied:
I meant is why do you need to search this /usr/src/./linux-headers-tegra-oot* path at all?

hiteshkumar-nv replied:
Ohh, We have differ path for Nvidia OOT on AGX and IGX. So we need to check for both path.
AGX: /usr/src/nvidia/nvidia-oot/
IGX: /usr/src/./linux-headers-tegra-oot

pateldipen1984-nv commented:
why name changed to build-for-jetson-drive-igpu-native.sh? I think original name was perfect.

hiteshkumar-nv committed:
Updated variable name and messages.

hiteshkumar-nv commented:
> why name changed to build-for-jetson-drive-igpu-native.sh? I think original name was perfect.

I kept file name as it is https://github.com/NVIDIA/jetson-rdma-picoevb/blob/master/kernel-module/build-for-jetson-drive-igpu-native.sh

pateldipen1984-nv commented on kernel-module/build-for-jetson-drive-igpu-native.sh:
```diff
@@ -20,22 +20,22 @@
+	echo "Nvdia OOT source not found in $NVIDIA_OOT_SRC_DIR"
+	NVIDIA_OOT_SRC_DIR="/usr/src/./linux-headers-tegra-oot*"
+	if [ ! -d ${NVIDIA_OOT_SRC_DIR} ]; then
+		echo "Nvidia OOT  source not found in $NVIDIA_OOT_SRC_DIR"
```
Only have echo "Nvidia OOT source not found"

hiteshkumar-nv replied:
Updated debug message. 👍

hiteshkumar-nv committed:
Updated debug messages.
